src/MICore/Transports/PipeTransport.cs

```diff
@@ -35,6 +36,14 @@ protected override string GetThreadName()
             return "MI.PipeTransport";
         }
 
+        public override int DebuggerPid
+        {
+            get
+            {
+                return _debuggerPid;
```

Comment:
`PipeTransport` is used when `useExternalConsole = false` even for local OS X and Linux, so we still need the pid in that case.

I haven't looked into when `TcpTransport` is used, but it doesn't seem like there's any access to any type of local process in that case.